deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ErrorCodeDescription.java

```diff
@@ -6,73 +6,181 @@
+     * disabled. May be thrown by operations such as getting a module by id.
+     */
     ModuleNotFound,
+
+    /**
+     * There's already a device with the same device Id in the IoT hub. This can be returned on by the service when adding
+     * a device to your IoT hub with an Id that is already in use.
+     */
     DeviceAlreadyExists,
+
+    /**
+     * The operation failed because it attempted to add a module to a device when that device already has a module
+     * registered to it with the same Id. This issue can be fixed by removing the existing module from the device first.
+     */
     ModuleAlreadyExistsOnDevice,
-    DeviceMessageLockLost,
+
+    /**
+     * When the message that you attempted to send is too large for IoT hub you will receive this error.
+     * @see <a href="https://aka.ms/iothubthrottling#other-limits">More details on message size limits</a>
+     */
     MessageTooLarge,
+
+    /**
+     * Too many devices were included in the bulk operation.
+     * @see <a href="https://docs.microsoft.com/rest/api/iothub/service/bulk-registry/update-registry">More details on device count limits</a>
+     */
     TooManyDevices,
-    TooManyModulesOnDevice,
+
+    /**
+     * IoT hub throttling limits have been exceeded for the requested operation.
+     * @see <a href="https://aka.ms/iothubthrottling">More details on throttling limits.</a>
+     */
     ThrottleBacklogLimitExceeded,
-    InvalidThrottleParameter,
+
+    /**
+     * IoT hub encountered a service-side issue. There can be a number of causes for a 500xxx error response.
+     * In all cases, the issue is most likely transient. IoT hub nodes can occasionally experience transient faults.
+     * When your application tries to connect to a node that is having issues, you receive this error.
+     * To mitigate 500xxx errors, issue a retry from your application.
+     */
     ServerError,
-    JobCancelled,
-    ConnectionForcefullyClosedOnNewConnection,
+
+    /**
+     * IoT hub failed to invoke the direct method because the target device was not connected at the time. To resolve
+     * this issue, you will need to make sure that your device is connected.
+     */
     DeviceNotOnline,
-    DeviceConnectionClosedRemotely,
 
     // These descriptions belong to 3 digit http response codes such as 404.
+    /**
+     * The general error for malformed service requests.
+     */
     BadFormat,
+
+    /**
+     * The general error for unauthorized service requests.
+     */
     Unauthorized,
+
+    /**
+     * The general error for forbidden service requests.
+     */
     Forbidden,
+
+    /**
+     * The general error for service requests that failed because a resource could not be found.
+     */
     NotFound,
+
+    /**
+     * The general error for service requests that failed because the resource already exists.
+     */
     Conflict,
 
     /**
      * Represents status code for 429 and for status code 429001
```

Comment:
we can also update this to call out that this happens during etag mismatch

Reply:
I'll also fix that this is supposed to be for 412, not 429. I'm not sure why this comment says 429 here